clash-starters/clash-example-project/clash-example-project.cabal

```diff
@@ -0,0 +1,103 @@
+cabal-version:       2.4
+name:                clash-example-project
+version:             0.1
+license-file:        LICENSE
+author:              Example User
+maintainer:          user@example.com
+
+common common-options
+  default-extensions:
+    BangPatterns
+    BinaryLiterals
+    ConstraintKinds
+    DataKinds
+    DefaultSignatures
+    DeriveAnyClass
+    DeriveDataTypeable
+    DeriveFoldable
+    DeriveFunctor
+    DeriveGeneric
+    DeriveLift
+    DeriveTraversable
+    DerivingStrategies
+    InstanceSigs
+    KindSignatures
+    LambdaCase
+    NoStarIsType
+    PolyKinds
+    RankNTypes
+    ScopedTypeVariables
+    StandaloneDeriving
+    TupleSections
+    TypeApplications
+    TypeFamilies
+    TypeOperators
+    ViewPatterns
+
+    -- TemplateHaskell is used to support convenience functions such as
+    -- 'listToVecTH' and 'bLit'.
+    TemplateHaskell
+    QuasiQuotes
+
+    -- Prelude isn't imported by default as Clash offers Clash.Prelude
+    NoImplicitPrelude
```

Comment:
Cabal since 2.0 supports mixins, I would recommend

```cabal
mixins: clash-prelude (Clash.Prelude as Prelude)
```

and leave `-XImplicitPrelude` on. This way `Prelude` is automatically replaced with `Clash.Prelude` for Clash projects.

Reply:
Ah that's cool, didn't know about that. I'm hesitant though, as you'd get a split between Clash code (src) and Haskell code (tests). This might be non-obvious for beginners.

Reply:
@christiaanb What do you think?

Reply:
Ah, mixins are a backpack feature, no? How well does that work with Stack and/or Nix? Anyhow, I'd rather keep the NoImplicitPrelude for now, and just use explicit imports. At least for this first release of the starter project.

Reply:
Backpack is not supported by Stack. Let's keep `NoImplicitPrelude` for now.